src/main/java/dev/projectg/geyserhub/GeyserHubMain.java

```diff
@@ -1,151 +1,103 @@
-            Placeholders.PAPI = 0;
-        }
-        if (this.getServer().getPluginManager().getPlugin("Essentials") == null) {
-            Placeholders.essentials = 0;
-        } else {
-            this.setupEconomy();
-        }
-        this.Scheduler();
-        Bukkit.getServer().getPluginManager().registerEvents(new ScoreboardManager(), this);
+        // The random interval broadcast module
+        Broadcast.startBroadcastTimer(getServer().getScheduler());
     }
 
-    public boolean loadConfiguration() {
-        File configFile = new File(getDataFolder(), "config.yml");
-        if (!configFile.exists()) {
-            try {
-                configFile.getParentFile().mkdirs();
-            } catch (SecurityException e) {
-                e.printStackTrace();
-                return false;
-            }
-            saveResource("config.yml", false);
-        }
-        // Get the config but don't actually load it into the main memory config
-        FileConfiguration config = new YamlConfiguration();
-        try {
-            config.load(configFile);
-            if (!config.contains("Config-Version", true)) {
-                logger.severe("Config-Version does not exist!");
-                return false;
-            } else if (!config.isInt("Config-Version")) {
-                logger.severe("Config-Version is not an integer!");
-                return false;
-            } else if (!(config.getInt("Config-Version") == configVersion)) {
-                logger.severe("Mismatched config version!");
-                return false;
-            } else {
-                reloadConfig();
-                return true;
-            }
-        } catch (IOException | InvalidConfigurationException e) {
-            e.printStackTrace();
-            return false;
-        }
-    }
-    public void Scheduler() {
+    public void initializeScoreboard() {
         Bukkit.getScheduler().scheduleSyncRepeatingTask(this, () -> {
             try {
                 ScoreboardManager.addScoreboard();
             } catch (Exception var2) {
                 var2.printStackTrace();
             }
-        }, 20L, Placeholders.isb * 20L);
-    }
-
-    // todo: I think we can just set vault as a softdepend instead of doing this?
-
-    private void setupEconomy() {
-        RegisteredServiceProvider economyProvider = getServer().getServicesManager().getRegistration(Economy.class);
-        if (economyProvider != null) {
-            Placeholders.economy = (Economy) economyProvider.getProvider();
-        }
-    }
-
-    private void setupPermissions() {
-        RegisteredServiceProvider permissionProvider = this.getServer().getServicesManager().getRegistration(Permission.class);
-        if (permissionProvider != null) {
-            Placeholders.permission = (Permission) permissionProvider.getProvider();
-        }
+        }, 20L, ScoreboardManager.REFRESH_RATE * 20L);
```

Comment:
Start schedulers for different modules regardless of config (and handle config within), or only start schedule if enabled in the config